### PR TITLE
Make danger.github.api available in event runs

### PR DESCRIPTION
**Pull request by jtreanor**
Currently non-PR runs do not have `danger.github.api` set. This manually adds it using the existing `octokitAPIForPeril` function.

### CODE REVIEW

File: api/source/github/events/handlers/_tests/fixtures/PerilRunnerEventBootStrapExample.json [hunk 1]
```diff
@@ -243,6 +243,36 @@
...
+          "pulls": {},
+          "rateLimit": {},
+          "reactions": {},
+          "repos": {},
+          "search": {},
+          "teams": {},
+          "users": {},
+          "gitdata": {},
+          "authorization": {},
+          "pullRequests": {}
```
Comment (orta) [inline api/source/github/events/handlers/_tests/fixtures/PerilRunnerEventBootStrapExample.json]: letting this into the snapshot is probably a bad idea (brittle)

File: api/source/github/events/handlers/event.ts [hunk 2]
```diff
@@ -42,14 +43,22 @@ export const runEventRun = async (
...
   }
 
+  // Manually set the Octokit API object so that danger.github.api works
+  const dangerDSL = {
+    github: {
+      ...event,
+      api: await octokitAPIForPeril(settings.installationID, token),
+    },
+  }
+
```
Comment (orta) [inline api/source/github/events/handlers/event.ts]: But I'm totally cool with this, it's a good idea IMO

If you replace the GitHub API in the snapshot with some dumb object (yeah, it kinda undermines the test but I'd rather that then a test which mysteriously breaks later)

**Comment by jtreanor** [comment]
Yeah, I totally agree having it in the snapshots isn't ideal. I have updated them now.

File: api/source/github/events/handlers/_tests/_events-create-fixture.test.ts [hunk 3]
```diff
@@ -44,10 +45,15 @@ it("passes the right args to the hyper functions", async () => {
...
 
-  // Take the payload, remove the JWT and save a copy of the JSON into a fixture dir, then snapshot it
+  // Take the payload, remove the JWT and Github api and save a copy of the JSON into a fixture dir, then snapshot it
   const payload = (runFromSameHost as any).mock.calls[0][0] as PerilRunnerBootstrapJSON
   payload.perilSettings.perilJWT = "12345"
   payload.perilSettings.perilRunID = "[run-id]"
+
+  // Ensure Github API is set before clearing it for the snapshot
+  expect((payload.payload.dsl as DangerDSLJSONType).github!.api).toBeDefined()
+  delete (payload.payload.dsl as DangerDSLJSONType).github!.api
```
Comment (orta) [inline api/source/github/events/handlers/_tests/_events-create-fixture.test.ts]: this is perfect!

**Pull request merged**